bin/p2-inspect/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/armon/consul-api"
+	"github.com/square/p2/pkg/intent"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/reality"
+	"gopkg.in/alecthomas/kingpin.v1"
+)
+
+var (
+	consulUrl      = kingpin.Flag("consul", "The hostname and port of a consul agent in the p2 cluster. Defaults to 0.0.0.0:8500.").String()
+	filterNodeName = kingpin.Flag("node", "The node to inspect. By default, all nodes are shown.").String()
+	filterPodId    = kingpin.Flag("pod", "The pod manifest ID to inspect. By default, all pods are shown.").String()
+)
+
+const (
+	INTENT_SOURCE = iota
+	REALITY_SOURCE
+)
+
+type NodePodStatus struct {
+	NodeName           string      `json:"node,omitempty"`
+	PodId              string      `json:"pod,omitempty"`
+	IntentManifestSHA  string      `json:"intent_manifest_sha"`
+	RealityManifestSHA string      `json:"reality_manifest_sha"`
+	Health             *NodeHealth `json:"health_check,omitempty"`
+}
+
+type NodeHealth struct {
+	Status string `json:"status"`
+	Output string `json:"output"`
+}
+
+func main() {
+	kingpin.Version("0.0.1")
+	kingpin.Parse()
+
+	client, err := consulapi.NewClient(&consulapi.Config{
+		Address: *consulUrl,
+	})
+	if err != nil {
+		log.Fatalf("Could not open consul client: %s", err)
+	}
+
+	intents, _, err := client.KV().List(intent.INTENT_TREE, nil)
+	if err != nil {
+		log.Fatalf("Could not list intent kvpairs: %s", err)
+	}
+	realities, _, err := client.KV().List(reality.REALITY_TREE, nil)
+	if err != nil {
+		log.Fatalf("Could not list reality kvpairs: %s", err)
+	}
+
+	statusMap := make(map[string]map[string]NodePodStatus)
+
+	for _, kvp := range intents {
+		if addKVPToMap(kvp, INTENT_SOURCE, *filterNodeName, *filterPodId, statusMap) != nil {
+			log.Fatal(err)
+		}
+	}
+
+	for _, kvp := range realities {
+		if addKVPToMap(kvp, REALITY_SOURCE, *filterNodeName, *filterPodId, statusMap) != nil {
+			log.Fatal(err)
+		}
+	}
+
+	for podId := range statusMap {
+		checks, _, err := client.Health().Checks(podId, nil)
```

Comment:
I'd like to see some of this make its way to the `health` package at some point, at least the Consul part. Part of what this command does is already implemented in the replicator, although definitely with the goal of being informational.